src/otx/api/entities/datasets.py

```diff
@@ -349,10 +349,14 @@ def get_combined_subset(self, subsets: List[Subset]) -> "DatasetEntity":
         Returns:
             DatasetEntity: DatasetEntity with items matching subsets
         """
-        dataset = DatasetEntity(
-            items=[item for item in self._items if item.subset in set(subsets)],
-            purpose=self.purpose,
-        )
+
+        dataset: DatasetEntity = DatasetEntity(purpose=self.purpose)
+
+        if subsets:
+            dataset = self.get_subset(subsets[0])
+            for subset in subsets[1:]:
+                dataset += self.get_subset(subset)
```

Comment:
What's is the difference with the following?

```suggestion
        if subsets:
             dataset = DatasetEntity(
                 items=[item for item in self._items if item.subset in set(subsets)],
                 purpose=self.purpose,
             )
        else:
             dataset: DatasetEntity = DatasetEntity(purpose=self.purpose)
```

Repeating `get_subset()` needs `O(m*n)` (`m` is the number of subsets and `n` is the number of dataset items).

Reply:
The difference is on sc_sdk side: get_subset is overloaded there, that's why this loop resulted in an empty dataset in Geti envoronment